pallets/foreign-investments/src/impls.rs

```diff
@@ -209,23 +213,66 @@ impl<T: Config> ForeignInvestment<T::AccountId> for Pallet<T> {
 	}
 }
 
+impl<T: Config> Pallet<T> {
+	fn deposit_apply_swap_events(
+		who: &T::AccountId,
+		swap_id: SwapId<T>,
+		swap: &SwapOf<T>,
+		status: &SwapStatus<T::SwapBalance>,
+	) -> DispatchResult {
+		if !status.swapped.is_zero() {
+			Pallet::<T>::deposit_event(Event::SwapCancelled {
+				who: who.clone(),
+				swap_id,
+				remaining: Swap {
+					amount_out: status.pending,
+					..swap.clone()
+				},
+				cancelled_in: status.swapped,
+				opposite_in: T::Swaps::pending_amount(who, swap_id, swap.currency_in)?,
+			});
+		}
+
+		if !status.pending.is_zero() {
+			Pallet::<T>::deposit_event(Event::SwapCreated {
+				who: who.clone(),
+				swap_id,
+				swap: Swap {
+					amount_out: status.pending,
+					..swap.clone()
+				},
+			})
+		}
```

Comment:
Could it be that both `status.swapped` and `status.pending` are non zero? 

IF YES: 
* How would one interpret the `swap_id` of the two events?

Reply:
Yes, it could. `swap_id` will be the same for both. Note that `swap_id` is just a tuple `(T::InvestmentId, Action)` not an `order_id` (which is a number from `order-book`).

If the above two events are dispatched, it means that when you apply a swap, part of the amount was automatically fulfilled by cancelation with an inverse order (notified by `SwapCancelled` event), and part of the swap could not be fulfilled with the inverse order: such remaining amount needs to be swapped (notified it by `SwapCreated` event).

Possible states after applying swaps are:
- `pending_amount == 0` and `swapped amount > 0` => it means applying a swap has partially or fully canceled any previous inverse swaps. The required amount is obtained from the inverse amount. No more amount needs to be swapped. OrderBook will Not create a new order.
- `pending amount > 0` and `swapped amount > 0` => It means applying a swap has fully canceled any previous inverse swaps, but more amount needs to be swapped. Orderbook will create a new order.
- `pending amount > 0` and `swapped amount == 0` => There was no inverse order, so we need to create a new order.


NOTE: Probably, a better name for `swap_status.swapped` is `swap_status.canceled`. It all depends on how you interpret it.

Reply:
Thanks a lot, that helped!